src/sentry/api/endpoints/project_codeowners.py

```diff
@@ -1,11 +1,144 @@
+import logging
+
+from django.db import IntegrityError
+from django.utils import timezone
+
+from rest_framework import serializers, status
 from rest_framework.response import Response
 
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.serializers import serialize
-from sentry.models import ProjectCodeOwners
+from sentry.models import (
+    ProjectCodeOwners,
+    ExternalUser,
+    ExternalTeam,
+    RepositoryProjectPathConfig,
+    User,
+)
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+from sentry.ownership.grammar import (
+    parse_code_owners,
+    convert_codeowners_syntax,
+)
+
+from sentry.api.endpoints.project_ownership import ProjectOwnershipSerializer, ProjectOwnershipMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectCodeOwnerSerializer(CamelSnakeModelSerializer):
+    code_mapping_id = serializers.IntegerField(required=True)
+    raw = serializers.CharField(required=True)
+    organization_integration_id = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = ProjectCodeOwners
+        fields = ["raw", "code_mapping_id", "organization_integration_id"]
+
+    def validate(self, attrs):
+        # If it already exists, set default attrs with existing values
+        if self.instance:
+            attrs = {
+                "raw": self.instance.raw,
+                "code_mapping_id": self.instance.repository_project_path_config,
+                **attrs,
+            }
+
+        if not attrs.get("raw", "").strip():
+            return attrs
+        external_association_err = ""
+        # Get list of team/user names from CODEOWNERS file
+        teamnames, usernames, emails = parse_code_owners(attrs["raw"])
+
+        # Check if there exists Sentry users with the emails listed in CODEOWNERS
+        users = User.objects.filter(email__in=emails)
```

Comment:
@NisanthanNanthakumar shouldn't we check the `UserEmail` table instead? That's generally we check for email matches in Senty.